src/WebJobs.Script/Diagnostics/MetricEventNames.cs

```diff
@@ -17,9 +17,35 @@ public static class MetricEventNames
         public const string HostStartupGrpcServerLatency = "host.startup.outofproc.grpcserver.initialize.latency";
         public const string HostStartupRuntimeLanguage = "host.startup.runtime.language.{0}";
 
+        // Script host level events
+        public const string ScriptHostManagerBuildScriptHost = "scripthostmanager.buildscripthost.latency";
+        public const string ScriptHostManagerStartScriptHost = "scripthostmanager.startscripthost.latency";
+        public const string ScriptHostManagerStartService = "scripthostmanager.start.latency";
+        public const string ScriptHostManagerRestartService = "scripthostmanager.restart.latency";
+
         // language worker level events
         public const string WorkerInitializeLatency = "host.startup.outofproc.{0}worker.initialize.attempt{1}.latency";
 
+        // FunctionMetadataprovider events
+        public const string ReadFunctionsMetadata = "functionmetadataprovider.readfunctionsmetadata";
+        public const string ReadFunctionMetadata = "functionmetadataprovider.readfunctionsmetadata.readfunctionmetadata";
+
+        // Host json file configuration events
+        public const string LoadHostConfigurationSource = "hostjsonfileconfigurationsource.loadhostconfigurationsource";
+        public const string LoadHostConfiguration = "hostjsonfileconfigurationsource.loadhostconfigurationsource.loadhostconfig";
+        public const string InitializeHostConfiguration = "hostjsonfileconfigurationsource.loadhostconfigurationsource.initializehostconfig";
+
+        // LanguageWorkerChannel events
+        public const string FunctionLoadRequestResponse = "languageworkerchannel.functionloadrequestresponse";
```

Comment:
Does the FunctionLoadRequestResponse actually track function being loaded by the language worker?

Reply:
Yup! It gets initialized at LoadRequest and gets disposed at LoadResponse. I've seen it take from 900 ms to over a second.

---
In reply to: [333798807](https://github.com/Azure/azure-functions-host/pull/4973#discussion_r333798807) [](ancestors = 333798807)